robot_smach_states/src/robot_smach_states/designator_iterator.py

```diff
@@ -60,11 +60,17 @@ def __init__(self, collection_des, element_des):
     def execute(self, userdata=None):
         if self._current_elements is None:
             collection = self.collection_des.resolve()
+            if collection is None:
+                rospy.loginfo("Collection is None".format(collection))
+                return 'stop_iteration'
+
             rospy.loginfo("Current elements: {}".format(collection))
             self._current_elements = iter(collection)
 
         try:
-            self.element_des.write(next(self._current_elements))
+            next_elem = next(self._current_elements)
+            rospy.loginfo("Iterate to next element {}".format(next_elem))
```

Comment:
I would change this to a logdebug. What do you think @jlunenburg

Reply:
I think the iterator is too low-level for a loginfo. We should be able to assume the iterator works as intended.